libraries/localization-utilities/src/TypingsGenerator.ts

```diff
@@ -62,7 +89,31 @@ export class TypingsGenerator extends StringValuesTypingsGenerator {
           });
         }
 
-        return { typings };
+        let exportAsDefaultInterfaceName: string | undefined;
+        if (inferDefaultExportInterfaceNameFromFilename) {
+          const lastSlashIndex: number = Math.max(filePath.lastIndexOf('/'), filePath.lastIndexOf('\\'));
+          const extensionIndex: number = Math.max(
+            filePath.lastIndexOf('.resx'),
+            filePath.lastIndexOf('.resjson'),
```

Comment:
May be faster to use endsWith and subtracting the extension length from the file path length

Reply:
Use `lastIndexOf('.')` and if the resulting extension appears to be `.json`, look again to see if we have `.resx.json` or `.loc.json`.